TeltonikaMonitor/src/main.ts

```diff
@@ -0,0 +1,780 @@
+    return (
+        map_camera.protocol != '' &&
+        map_camera.port != null &&
+        map_camera.ip != '' &&
+        map_camera.user != '' &&
+        map_camera.password != ''
+    );
+}
+
+//  ----------------------
+//  |  connect to modem  |
+//  ----------------------
+
+let portsInfo = [false, false, false, false, false];
+function transformSignalStrenght(strenght: number): number | never {
+    if (strenght > -57) {
+        return 5;
+    } else if (strenght > -67) {
+        return 4;
+    } else if (strenght > -82) {
+        return 3;
+    } else if (strenght > -97) {
+        return 2;
+    } else if (strenght > -111) {
+        return 1;
+    } else {
+        return 0;
+    }
+}
+function transformConnectionType(connection_type: string): string {
+    if (['5G-NSA', '5G-SA'].includes(connection_type)) {
+        return '5G';
+    } else if (['LTE', 'CAT-M1', 'FDD LTE', 'TDD LTE'].includes(connection_type)) {
+        return '4G';
+    } else if (
+        [
+            'WCDMA',
+            'HSDPA',
+            'HSUPA',
+            'HSPA',
+            'HSPA+',
+            'HSDPA and HSUPA',
+            'HSDPA and H',
+            'HSDPA+HSUPA',
+            'DC-HSPA+',
+        ].includes(connection_type)
+    ) {
+        return '3G';
+    } else if (['CDMA', 'EDGE', 'GPRS', 'GSM', 'CAT-NB1'].includes(connection_type)) {
+        return '2G';
+    } else {
+        return connection_type;
+    }
+}
+function getUpdateTime(date: string): string {
+    const d = new Date(date);
+    let hour = d.getHours().toString();
+    let minute = d.getMinutes().toString();
+    let second = d.getSeconds().toString();
+
+    if (hour.length == 1) {
+        hour = '0' + hour;
+    }
+    if (minute.length == 1) {
+        minute = '0' + minute;
+    }
+    if (second.length == 1) {
+        second = '0' + second;
+    }
+
+    return `${d.getDate()}. ${d.getMonth()}. ${d.getFullYear()} ${hour}:${minute}:${second}`;
+}
+function parseUptime(uptime: number): string {
+    let second = (uptime % 60).toString();
+    uptime = Math.floor(uptime / 60);
+
+    let minute = (uptime % 60).toString();
+    uptime = Math.floor(uptime / 60);
+
+    let hour = (uptime % 24).toString();
+    uptime = Math.floor(uptime / 24);
+
+    if (hour.length == 1) {
+        hour = '0' + hour;
+    }
+    if (minute.length == 1) {
+        minute = '0' + minute;
+    }
+    if (second.length == 1) {
+        second = `0` + second;
+    }
+
+    return `${uptime}:${hour}:${minute}:${second}`;
+}
+function parseResponse(
+    response: Record<string, unknown>,
+    wireless: { ssid: string; active: number }[],
+    ports: { id: number; name: string; type: string }[]
+): ModemInfo {
+    let wifi_2 = false;
+    let wifi_5 = false;
+
+    for (const { ssid, active } of wireless) {
+        if (ssid == 'RUT_C52E_2G' && active == 1) {
+            wifi_2 = true;
+        } else if (ssid == 'RUT_C52F_5G' && active == 1) {
```

Comment:
V celem skriptu pouzivej pokud to jde === a !==

Reply:
!